File: Dairy_Managemeny_System_Front_End/src/components/Product/ProductService.js
import axios from 'axios';
const base_url="http://localhost:7071/products/";
class ProductService{
    getAllProducts(){
       return axios.get(base_url+"product-list");
    }
    addProduct(prod){
        return axios.post(base_url+"add-product",prod,{headers:{
            'content-type':'application/json'}});
    }
    updateProduct(prod){
        return axios.put(base_url+"update-product",prod,{headers:{
            'content-type':'application/json'}});
    }
    deleteProduct(id){
        console.log("In Delete Method");
        return axios.delete(base_url+id);
    }

}

export default new ProductService();

File: Dairy_Management_System_Back-End/src/main/java/com/app/service/ProductService.java
package com.app.service;

import java.util.List;

import com.app.entities.Product;

public interface ProductService {
	
	List<Product> getAllProducts();

	Product addProduct(Product product);

	String deleteProduct(Long productId);

	Product updateProduct(Product product);

	Product getProductById(Long productId);

	void updateStockWithId(Long id, int newStock);
	
	//public List<Product> getAllProducts();
}
